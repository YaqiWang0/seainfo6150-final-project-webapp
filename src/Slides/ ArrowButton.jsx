import React from "react"
import style from "./ArrowButton.module.css"

const ArrowButton = (props) => {
    return (
        <a href="#" className={props.isRight === true ? style.arrow_right : style.arrow_left} onClick={props.onClick}></a>
    )
}

export default ArrowButton