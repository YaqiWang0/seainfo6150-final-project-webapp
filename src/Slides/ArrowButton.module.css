

.arrow_left {
  top: 50%;
  width: 3vmin;
  height: 3vmin;
  background: transparent;
  border-top: 1vmin solid white;
  border-right: 1vmin solid white;
  box-shadow: 0 0 0 lightgray;
  transition: all 200ms ease;
  left: 0;
  transform: translate3d(0, -50%, 0) rotate(-135deg);
}

.arrow_right {
  top: 50%;
  width: 3vmin;
  height: 3vmin;
  background: transparent;
  border-top: 1vmin solid white;
  border-right: 1vmin solid white;
  box-shadow: 0 0 0 lightgray;
  transition: all 200ms ease;
  right: 0;
  transform: translate3d(0,-50%,0) rotate(45deg);
}


.arrow_left:hover {
    border-color: darkred;
    box-shadow: 0.5vmin -0.5vmin 0 white;
  }

.arrow_right:hover {
  border-color: darkred;
  box-shadow: 0.5vmin -0.5vmin 0 white;
}

.arrow_left:before {
  content: '';
  top: 50%;
  left: 50%;
  transform: translate(-40%,-60%) rotate(45deg);
  width: 200%;
  height: 200%;
}

.arrow_right:before {
  content: '';
  top: 50%;
  left: 50%;
  transform: translate(-40%,-60%) rotate(45deg);
  width: 200%;
  height: 200%;
}