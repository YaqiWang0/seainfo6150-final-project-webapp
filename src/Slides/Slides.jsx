import React, {useState} from "react"
import style from "./Slides.module.css"
import ArrowButton from "./ ArrowButton";

const Slides = (props) => {

    const [select, setSelect] = useState(0);
    const [backgroundImage, setBackGroundImage] = useState({
                                                    backgroundImage: 'url(' + props.news[0].image + ')'
                                                })

    const clickDot = (e, index) => {
        e.stopPropagation()
        setSelect(index);
        setBackGroundImage({backgroundImage: 'url(' + props.news[select + 5].image + ')'})
        console.log(backgroundImage)
    }

    const clickButton = (e, isRight) => {
        e.stopPropagation()
        if(isRight === true) {
            if(select === 4) {
                setSelect(0)
            } else {
                setSelect(select + 1)
            }
        } else {
            if(select === 0) {
                setSelect(4)
            } else {
                setSelect(select - 1)
            }
        }

        setBackGroundImage({backgroundImage: 'url(' + props.news[select].image + ')'})
    }

    const click = () => {
        window.open(props.news[select].url, "_blank")
    }


return (
    <div className={style.container} style={backgroundImage} onClick={click}>
        <div className={style.cover}>
            <div className={style.left_arrow}>
                <ArrowButton isRight = {false} onClick = {(e) => clickButton(e, false)}/>
            </div>
            <div className={style.right_arrow}>
                <ArrowButton isRight = {true} onClick = {(e) => clickButton(e, true)}/>
            </div>
            <div className={style.text}>{props.news[select].title}</div>
            <ul className={style.dots}>
                <li  className={select == 0 ? style.dotSelected : style.dotUnSelected} onClick={(e) => clickDot(e, 0)}/>
                <li  className={select == 1 ? style.dotSelected : style.dotUnSelected} onClick={(e) => clickDot(e, 1)}/>
                <li  className={select == 2 ? style.dotSelected : style.dotUnSelected} onClick={(e) => clickDot(e, 2)}/>
                <li  className={select == 3 ? style.dotSelected : style.dotUnSelected} onClick={(e) => clickDot(e, 3)}/>
                <li  className={select == 4 ? style.dotSelected : style.dotUnSelected} onClick={(e) => clickDot(e, 4)}/>
            </ul>
        </div>
   </div>

)
}

export default Slides