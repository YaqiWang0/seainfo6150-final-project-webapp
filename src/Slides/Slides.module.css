.dotSelected {
    height: 1.5vw;
    width: 1.5vw;
    background-color: #bbb;
    border-radius: 50%;
    display: inline-block;
    margin: 0.3vw;
    cursor: pointer;
}

.dotUnSelected {
    height: 1vw;
    width: 1vw;
    background-color: rgba(187, 187, 187, 0.49);
    border-radius: 50%;
    display: inline-block;
    margin: 0.3vw;
    cursor: pointer;
}

.container {
    background-size: cover;
    width: 100%;
    height: 100%;
    cursor: pointer;
}

.dots {
    width: 100%;
    display: flex;
    flex-direction: row;
    grid-area: footer;
    align-items: center;
    justify-content: center;
}

.left_arrow {
    grid-area: leftArrow;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
}

.right_arrow {
    grid-area: rightArrow;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
}

.text {
    grid-area: title;
    display: flex;
    flex-direction: row;
    align-items: flex-end;
    justify-content: center;
    color: white;
    text-align: left;
    font-size: 2vw;
    padding-right: 3vw;
}

.cover {
    width: 100%;
    height: 100%;
    background-color: rgba(0, 0, 0, 0.59);
    background-size: cover;
    display: grid;
    height: 100%;
    grid-template-columns: 1fr 3fr 1fr 3fr 1fr;
    grid-template-rows: 4fr 1fr;
    grid-template-areas:
            "leftArrow title . . rightArrow"
            "leftArrow footer footer footer rightArrow";
}