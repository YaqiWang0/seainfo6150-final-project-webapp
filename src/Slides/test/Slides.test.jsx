import React from "react";
import data from "./data"
import Slides from "../Slides";
import {fireEvent} from "@testing-library/react";

describe("Slides test", () => {

    it('render successfully', function () {
        const {container} = render(
            <Slides news = {data}/>
        );
        expect(container).toMatchSnapshot();
    })

    it('render successfully when click left arrow', function () {
        const {container, getByRole} = render(
            <Slides news = {data}/>
        );
        fireEvent.click(getByRole("left"));
        expect(container).toMatchSnapshot();
    })

    it('render successfully when click right arrow', function () {
        const {container, getByRole} = render(
            <Slides news = {data}/>
        );
        fireEvent.click(getByRole("right"));
        expect(container).toMatchSnapshot();
    })

    it('render successfully when click dot', function () {
        const {container, getByRole} = render(
            <Slides news = {data}/>
        );
        fireEvent.click(getByRole("dot"));
        expect(container).toMatchSnapshot();
    })
})