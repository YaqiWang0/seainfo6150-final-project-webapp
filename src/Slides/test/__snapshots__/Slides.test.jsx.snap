// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsItem test render successfully 1`] = `
<div>
  <div
    class="container"
    style="background-image: url(https://cdn.snapi.dev/images/v1/h/h/bussines-services9.jpg);"
  >
    <div
      class="cover"
    >
      <div
        class="left_arrow"
        role="left"
      >
        <a
          class="arrow_left"
          href="#"
        />
      </div>
      <div
        class="right_arrow"
        role="right"
      >
        <a
          class="arrow_right"
          href="#"
        />
      </div>
      <div
        class="text"
      >
        5 Stocks Growing Earnings
      </div>
      <ul
        class="dots"
      >
        <li
          class="dotSelected"
          role="dot"
        />
        <li
          class="dotUnSelected"
        />
        <li
          class="dotUnSelected"
        />
        <li
          class="dotUnSelected"
        />
        <li
          class="dotUnSelected"
        />
      </ul>
    </div>
  </div>
</div>
`;

exports[`NewsItem test render successfully when click dot 1`] = `
<div>
  <div
    class="container"
    style="background-image: url(https://cdn.snapi.dev/images/v1/s/o/so2kdddd.);"
  >
    <div
      class="cover"
    >
      <div
        class="left_arrow"
        role="left"
      >
        <a
          class="arrow_left"
          href="#"
        />
      </div>
      <div
        class="right_arrow"
        role="right"
      >
        <a
          class="arrow_right"
          href="#"
        />
      </div>
      <div
        class="text"
      >
        5 Stocks Growing Earnings
      </div>
      <ul
        class="dots"
      >
        <li
          class="dotSelected"
          role="dot"
        />
        <li
          class="dotUnSelected"
        />
        <li
          class="dotUnSelected"
        />
        <li
          class="dotUnSelected"
        />
        <li
          class="dotUnSelected"
        />
      </ul>
    </div>
  </div>
</div>
`;

exports[`NewsItem test render successfully when click left arrow 1`] = `
<div>
  <div
    class="container"
    style="background-image: url(https://cdn.snapi.dev/images/v1/h/h/bussines-services9.jpg);"
  >
    <div
      class="cover"
    >
      <div
        class="left_arrow"
        role="left"
      >
        <a
          class="arrow_left"
          href="#"
        />
      </div>
      <div
        class="right_arrow"
        role="right"
      >
        <a
          class="arrow_right"
          href="#"
        />
      </div>
      <div
        class="text"
      >
        Becton, Dickinson & Co. Is a Highly Profitable, Recession-Proof Company
      </div>
      <ul
        class="dots"
      >
        <li
          class="dotUnSelected"
          role="dot"
        />
        <li
          class="dotUnSelected"
        />
        <li
          class="dotUnSelected"
        />
        <li
          class="dotUnSelected"
        />
        <li
          class="dotSelected"
        />
      </ul>
    </div>
  </div>
</div>
`;

exports[`NewsItem test render successfully when click right arrow 1`] = `
<div>
  <div
    class="container"
    style="background-image: url(https://cdn.snapi.dev/images/v1/h/h/bussines-services9.jpg);"
  >
    <div
      class="cover"
    >
      <div
        class="left_arrow"
        role="left"
      >
        <a
          class="arrow_left"
          href="#"
        />
      </div>
      <div
        class="right_arrow"
        role="right"
      >
        <a
          class="arrow_right"
          href="#"
        />
      </div>
      <div
        class="text"
      >
        Pfizer Analyst Breaks Down The FDA Process For Coronavirus Vaccine Candidate
      </div>
      <ul
        class="dots"
      >
        <li
          class="dotUnSelected"
          role="dot"
        />
        <li
          class="dotSelected"
        />
        <li
          class="dotUnSelected"
        />
        <li
          class="dotUnSelected"
        />
        <li
          class="dotUnSelected"
        />
      </ul>
    </div>
  </div>
</div>
`;

exports[`Slides test render successfully 1`] = `
<div>
  <div
    class="container"
    style="background-image: url(https://cdn.snapi.dev/images/v1/h/h/bussines-services9.jpg);"
  >
    <div
      class="cover"
    >
      <div
        class="left_arrow"
        role="left"
      >
        <a
          class="arrow_left"
          href="#"
        />
      </div>
      <div
        class="right_arrow"
        role="right"
      >
        <a
          class="arrow_right"
          href="#"
        />
      </div>
      <div
        class="text"
      >
        5 Stocks Growing Earnings
      </div>
      <ul
        class="dots"
      >
        <li
          class="dotSelected"
          role="dot"
        />
        <li
          class="dotUnSelected"
        />
        <li
          class="dotUnSelected"
        />
        <li
          class="dotUnSelected"
        />
        <li
          class="dotUnSelected"
        />
      </ul>
    </div>
  </div>
</div>
`;

exports[`Slides test render successfully when click dot 1`] = `
<div>
  <div
    class="container"
    style="background-image: url(https://cdn.snapi.dev/images/v1/s/o/so2kdddd.);"
  >
    <div
      class="cover"
    >
      <div
        class="left_arrow"
        role="left"
      >
        <a
          class="arrow_left"
          href="#"
        />
      </div>
      <div
        class="right_arrow"
        role="right"
      >
        <a
          class="arrow_right"
          href="#"
        />
      </div>
      <div
        class="text"
      >
        5 Stocks Growing Earnings
      </div>
      <ul
        class="dots"
      >
        <li
          class="dotSelected"
          role="dot"
        />
        <li
          class="dotUnSelected"
        />
        <li
          class="dotUnSelected"
        />
        <li
          class="dotUnSelected"
        />
        <li
          class="dotUnSelected"
        />
      </ul>
    </div>
  </div>
</div>
`;

exports[`Slides test render successfully when click left arrow 1`] = `
<div>
  <div
    class="container"
    style="background-image: url(https://cdn.snapi.dev/images/v1/h/h/bussines-services9.jpg);"
  >
    <div
      class="cover"
    >
      <div
        class="left_arrow"
        role="left"
      >
        <a
          class="arrow_left"
          href="#"
        />
      </div>
      <div
        class="right_arrow"
        role="right"
      >
        <a
          class="arrow_right"
          href="#"
        />
      </div>
      <div
        class="text"
      >
        Becton, Dickinson & Co. Is a Highly Profitable, Recession-Proof Company
      </div>
      <ul
        class="dots"
      >
        <li
          class="dotUnSelected"
          role="dot"
        />
        <li
          class="dotUnSelected"
        />
        <li
          class="dotUnSelected"
        />
        <li
          class="dotUnSelected"
        />
        <li
          class="dotSelected"
        />
      </ul>
    </div>
  </div>
</div>
`;

exports[`Slides test render successfully when click right arrow 1`] = `
<div>
  <div
    class="container"
    style="background-image: url(https://cdn.snapi.dev/images/v1/h/h/bussines-services9.jpg);"
  >
    <div
      class="cover"
    >
      <div
        class="left_arrow"
        role="left"
      >
        <a
          class="arrow_left"
          href="#"
        />
      </div>
      <div
        class="right_arrow"
        role="right"
      >
        <a
          class="arrow_right"
          href="#"
        />
      </div>
      <div
        class="text"
      >
        Pfizer Analyst Breaks Down The FDA Process For Coronavirus Vaccine Candidate
      </div>
      <ul
        class="dots"
      >
        <li
          class="dotUnSelected"
          role="dot"
        />
        <li
          class="dotSelected"
        />
        <li
          class="dotUnSelected"
        />
        <li
          class="dotUnSelected"
        />
        <li
          class="dotUnSelected"
        />
      </ul>
    </div>
  </div>
</div>
`;
